app/src/main/kotlin/com/exactpro/th2/lwdataprovider/entities/responses/CustomSerializer.kt

```diff
@@ -144,11 +144,33 @@ fun Event.toJSONByteArray(): ByteArray =
         write(CLOSING_CURLY_BRACE)
     }.toByteArray()
 
-private fun jsonEscape(value: String): String {
-    return if (ESCAPE_CHARACTERS.any(value::contains)) {
-        JsonEscapeUtil.escape(value)
-    } else {
-        value
+internal fun jsonEscape(value: String): String {
+    val noEscapeRequired = value.chars().noneMatch {
+        it.toChar().let { ch ->
+            ch in ESCAPE_CHARACTERS || ch < ' ' || ch == '\u007f'
+        }
+    }
+    if (noEscapeRequired) {
+        return value
+    }
+    return buildString(value.length) {
```

Comment:
the built text has more length than origin because this method adds character for escaping

Reply:
Yes, but this is just the initial capacity. We don't know how many characters will be escaped and how many chars the string will need. So, the best option here is to use the original size of the string, in my opinion